libs/h3/src/main/java/org/elasticsearch/h3/HexRing.java

```diff
@@ -581,52 +599,116 @@ public static long[] hexRing(long origin) {
         int idx = 0;
         long previous = -1;
         for (int i = 0; i < 6; i++) {
-            int[] rotations = new int[] { 0 };
-            long[] nextNeighbor = new long[] { 0 };
-            int neighborResult = h3NeighborRotations(origin, DIRECTIONS[i].digit(), rotations, nextNeighbor);
-            if (neighborResult != E_PENTAGON) {
-                // E_PENTAGON is an expected case when trying to traverse off of
+            long neighbor = h3NeighborRotations(origin, DIRECTIONS[i].digit());
```

Comment:
It looks like the original method must have been used in other cases where the rotations had some value. I wonder if we want to change the method name to reflect the new, simpler, behaviour?

Reply:
Changed the name to `h3NeighborInDirection`.

The difference is that the original method allowed to get the k-ring, being k the distance in H3bin of the ring. We only allow to get the first ring so it can be simplified.